ext/liquid_c/variable.c

```diff
@@ -3,6 +3,137 @@
 #include "parser.h"
 #include <stdio.h>
 
+static ID id_rescue_strict_parse_syntax_error;
+
+static void compile_expression(vm_assembler_t *code, VALUE expression, bool const_expression)
+{
+    if (const_expression)
+        vm_assembler_add_push_const(code, expression);
+    else
+        vm_assembler_add_push_eval_expr(code, expression);
+}
+
+static int compile_each_keyword_arg(VALUE key, VALUE value, VALUE func_arg)
+{
+    vm_assembler_t *code = (vm_assembler_t *)func_arg;
+
+    vm_assembler_add_push_const(code, key);
+
+    bool is_const_expr = rb_respond_to(value, id_evaluate) == Qtrue;
+    compile_expression(code, value, is_const_expr);
+
+    return ST_CONTINUE;
+}
+
+static inline void parse_and_compile_expression(parser_t *p, vm_assembler_t *code)
+{
+    bool is_const = will_parse_constant_expression_next(p);
```

Comment:
Is this mostly an optimization over checking for `#evaluate`?

Reply:
It was mostly a forward thinking approach, since we want to avoid even creating a VariableLookup or RangeLookup when parsing expressions in the future.  However, I did also think it would be faster than calling `rb_respond_to`